src/fdwave_raja/target_3d.cpp

```diff
@@ -200,33 +251,82 @@ void target_pml_3d(llint nx, llint ny,llint nz,
                   hdx_2, hdy_2, hdz_2,
                   coefxViewConst, coefyViewConst, coefzViewConst,
                   uViewConst, vpViewConst, etaViewConst, vView, phiView );
+#endif
+#if KERNEL_OPTION == 1
+  if( !haveRun )
+  {
+    haveRun = true;
+    std::cout << "Using RAJA kernel with pointers" << std::endl;
+  }
+
+  RAJA::RangeSegment const XRange(x3, x4);
+  RAJA::RangeSegment const YRange(y3, y4);
+  RAJA::RangeSegment const ZRange(z3, z4);
   
-  // const float coef0 = coefxViewConst(0) + coefyViewConst(0) + coefzViewConst(0);
-  // RAJA_UNUSED_VAR( nx );
-  
-  //   for (llint i = x3; i < x4; ++i) {
-  //      for (llint j = y3; j < y4; ++j) {
-  //          for (llint k = z3; k < z4; ++k) {
-  //              float const lap = LAP;
+  using POLICY = RAJA::KernelPolicy<
+    RAJA::statement::For<0, RAJA::loop_exec, 
+      RAJA::statement::For<1, RAJA::loop_exec, 
+        RAJA::statement::For<2, RAJA::loop_exec, 
+          RAJA::statement::Lambda<0>
+        >
+      >
+    >
+  >;
 
-  //              v[IDX3_l(i,j,k)] =
-  //                  ((2.f-eta[IDX3_eta1(i,j,k)]*eta[IDX3_eta1(i,j,k)]
-  //                  +2.f*eta[IDX3_eta1(i,j,k)])*u[IDX3_l(i,j,k)]
-  //                  -v[IDX3_l(i,j,k)]
-  //                  +vp[IDX3(i,j,k)]*
-  //                  (lap+phi[IDX3(i,j,k)]))/(1.f+2.f*eta[IDX3_eta1(i,j,k)]);
+  RAJA::kernel<POLICY>( RAJA::make_tuple(XRange, YRange, ZRange),
```

Comment:
Same loop body as the native one below, but with RAJA::kernel and it gets the same (within less than 5%) good performance.